File: ch7/PassArray.java
// PassArray.java - This program demonstrates passing an array to a method.
// Compile & run: javac PassArray.java && java PassArray

public class PassArray {
    static void vaTest(int[] v) {
        System.out.print("Number of args: " + v.length + " Contents: ");
        
        for (int x : v) {
            System.out.print(x + " ");
        }
        
        System.out.println();
    }
    
    public static void main(String[] args) {
        // Notice how we can pass an array to a method
        int[] n1 = {10};
        int[] n2 = {1, 2, 3};
        int[] n3 = {};
        
        vaTest(n1); // 1 arg
        vaTest(n2); // 3 args
        vaTest(n3); // 0 args
    }
    
}

File: ch7/PassArray.go
package main
import "fmt"

// PassArray.go - Pass an array to a function
// Compile & run: go run PassArray.go

func vaTest(v []int) {
    fmt.Printf("Number of args: %d Contents: ", len(v))
    
    for _, x := range v {
        fmt.Printf("%d ", x)
    }
    
    fmt.Println()
}

func main() {
    // Notice how we can pass an array to a function
    n1 := []int{10}
    n2 := []int{1, 2, 3}
    n3 := []int{}
    
    vaTest(n1) // 1 arg
    vaTest(n2) // 3 args
    vaTest(n3) // 0 args
}